src/main/java/org/apache/commons/imaging/formats/webp/chunks/WebPChunk.java

```diff
@@ -61,22 +74,41 @@ public String getTypeDescription() {
                 (byte) ((type >> 24) & 0xff)}, StandardCharsets.UTF_8);
     }
 
+    /**
+     * @return the payload size.
+     */
     public int getPayloadSize() {
         return size;
     }
 
+    /**
+     * @return the chunk size.
+     */
     public int getChunkSize() {
         // if chunk size is odd, a single padding byte is added
         int padding = (size % 2) != 0 ? 1 : 0;
 
         // Chunk FourCC (4 bytes) + Chunk Size (4 bytes) + Chunk Payload (n bytes) + Padding
-        return 4 + 4 + size + padding;
+        int chunkFourCCandSize = 4 + 4;
+        int paddedSize = Math.addExact(size, padding);
+        return Math.addExact(chunkFourCCandSize, paddedSize);
```

Comment:
I think we should check chunk size in constructor instead of here, this method should not throw exception.

Reply:
Good idea. Let me try that instead.

Reply:
Have a look if that's what you had in mind :+1: thanks!